### PR TITLE
chore: update guide to include path to public key for ssh signing

### PR DESCRIPTION
**Pull request by mitche50**
The config file for using SSH paths may not be parsed correctly on some systems.  Add in a hint in the documentation to give an alternative to use the direct path to your ssh pubkey file instead of the contents of your key.

### Why:

Closes [issue link]

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by cmwilson21** [comment]
@mitche50 Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by Amad3eu — COMMENTED** [review]
✅✅✅

**Comment by vtbassmatt** [comment]
:wave: hi from the product manager for Git Systems. Thanks for opening this PR!

I'm curious on what systems you've (or the internet at large has) seen a problem with using the literal key in this fashion. I couldn't find much with some searching, including on the Git mailing list, but I'm open to the idea that my search syntax was poor 😅 Could you send any pointers you have?

After doing some deep-diving on this topic, I have two alternate proposals for the docs team:
1. It looks like we could always suggest pointing to a file rather than a literal key, with no loss of functionality for our customers. Depending on what @mitche50 comes back with, we might want to change to that format and drop the literal syntax from our docs entirely.
2. Alternatively, we should emphasize the fact that you have to prefix your literal key with `key::`. That's not completely obvious as-written, and if you "merely" paste your key in without the prefix, you're relying on backcompat behavior in Git which may not be forwards-compatible with future key types.
Something like this should work, assuming the user correctly did the `pbcopy` step:
```
$ git config --global user.signingkey "key::$(pbpaste) user@example.com"
```
(NB I didn't actually test that syntax, I wrote it from memory.)

**Comment by mitche50** [comment]
The documentation (even with the `key::` prefix) did not work on Ubuntu 20.04.02 running on WSL2.  This may be relegated to just whatever internally is happening on the VM that the windows linux kernel is using, but WSL2 is prevalent enough to at least address it.

IIRC (this issue was created a while ago) I had colleagues who had done this successfully with the literal key on macbooks, unsure if I had anyone who was running bare metal linux distribution comment.

**Comment by vtbassmatt** [comment]
Thanks @mitche50, I appreciate the additional detail. What do you think about basically replacing our guidance to use a bare key with advice to use the file path (rather than having one, then the other as fallback)? From my read of the Git mailing list, that seems to be the preferred approach when the feature was designed.

**Comment by mitche50** [comment]
I think this works as well, the file should be compatible with all systems 👍

**Comment by timja** [comment]
I've verified on MacOS BigSur 

with:
```
❯ ssh -V
OpenSSH_9.0p1, OpenSSL 1.1.1q  5 Jul 2022
❯ git --version
git version 2.37.3
```

Using file path only means you don't need to pass the key inline or have an ssh agent running


```diff
diff --git a/config/git/config b/config/git/config
index 7aedf53..0df39fb 100755
--- a/config/git/config
+++ b/config/git/config
@@ -3,7 +3,7 @@
 # Please adapt and uncomment the following lines:
        name = Tim Jacomb
        useConfigOnly = true
-       signingkey = ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDnTfTt8lqJdcucKu538C6EvPNruiO9r67Jvfrbokear (omitted)\n
+       signingkey = ~/.ssh/id_ed25519.pub
```

**Review by Amad3eu — COMMENTED** [review]
Branch Conflict😣😣😣

**Comment by cmwilson21** [comment]
Hey @mitche50, are you up for making the changes to this PR that Matt suggested in the [previous comment](https://github.com/github/docs/pull/20172#issuecomment-1278034663)? No worries if not ✨

**Comment by mattpollard** [comment]
👋🏻 @mitche50, I'm planning on working on this early next week, but if you've got a chance and the interest before then, feel free to incorporate the changes that @vtbassmatt suggested ☺️

**Comment by japui-coder** [comment]
Confirm that the changes meet the user experience and goals outlined in the content design plan (if there is one).

File: data/reusables/gpg/paste-ssh-public-key.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 1. To set your SSH signing key in Git, paste the text below, substituting the contents of your clipboard for the key you'd like to use. Since the key contains spaces, you must wrap it in quotes:
   ```bash
   $ git config --global user.signingkey 'ssh-ed25519 AAAAC3(...) user@example.com'
+  ```
+
+Alternatively, you can add the direct path to your public key if your system does not properly handle adding the content of the key directly.
+  ```bash
+  $ git config --global user.signingkey=/path/to/.ssh/pubkey_filename.pub
   ```
```
Comment (mattpollard) [inline data/reusables/gpg/paste-ssh-public-key.md:9]:
```suggestion
1. To set your SSH signing key in Git, paste the text below, substituting **/PATH/TO/KEY.PUB** with the path to the pubic key you'd like to use.
  ```bash
  $ git config --global user.signingkey=/PATH/TO/.SSH/KEY.PUB
  ```
```

**Review by mattpollard — COMMENTED** [review]
@mitche50, I think this'll get the article in line with @vtbassmatt's suggestion. @vtbassmatt, could you also take a quick look?

Feel free to incorporate the change and then let us know when you're ready for us to merge. Otherwise, we'll get this merged in a couple of days if we don't hear back.

File: data/reusables/gpg/paste-ssh-public-key.md (hunk 1 above)
Comment (mitche50) [reply data/reusables/gpg/paste-ssh-public-key.md:9]: This LGTM - sorry have been distracted on my end - glad this is getting closed out!

**Comment by mattpollard** [comment]
Thanks, @mitche50 🙇🏻 We'll merge this at the end of the week after our freeze for [GitHub Universe 2022](https://githubuniverse.com) is over.

File: data/reusables/gpg/paste-ssh-public-key.md [hunk 2]
```diff
@@ -1,4 +1,3 @@
-1. To set your SSH signing key in Git, paste the text below, substituting the contents of your clipboard for the key you'd like to use. Since the key contains spaces, you must wrap it in quotes:
+1. To set your SSH signing key in Git, paste the text below, substituting **/PATH/TO/KEY.PUB** with the path to the pubic key you'd like to use.
```
Comment (timja) [inline data/reusables/gpg/paste-ssh-public-key.md:1]:
```suggestion
1. To set your SSH signing key in Git, paste the text below, substituting **/PATH/TO/KEY.PUB** with the path to the public key you'd like to use.
```

**Review by japui-coder — COMMENTED** [review]
data/reusables/gpg/paste-ssh-public-key.md . Patch.

**Review by cmwilson21 — APPROVED** [review]
:shipit:

**Comment by timja** [comment]
This is a nice improvement but it needs review against where the include is used, i.e. some docs don't make sense right now.

https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key

section 3, asks you to copy a key that isn't needed

**Comment by mattpollard** [comment]
@timja 👋🏻 Sorry, I lost track of this. The doc you links to looks alright to me, and doesn't seem to mention copying a key. Can you clarify?

**Comment by timja** [comment]
Seems to have been fixed, can't remember exactly what was there I should have taken a screenshot.

**Comment by Gojekgobiz** [comment]
> Seems to have been fixed, can't remember exactly what was there I should have taken a screenshot.

**Pull request merged**